jupyter/d2l-java/chapter_multilayer-perceptrons/dropout.ipynb

```diff
@@ -0,0 +1,812 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%loadFromPOM\n",
+    "<dependency>\n",
+    "    <groupId>tech.tablesaw</groupId>\n",
+    "    <artifactId>tablesaw-jsplot</artifactId>\n",
+    "    <version>0.30.4</version>\n",
+    "</dependency>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%load ../utils/plot-utils.ipynb\n",
+    "%load ../utils/DataPoints.java\n",
+    "%load ../utils/TrainingChapter4.java"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import java.nio.file.*;\n",
+    "\n",
+    "import ai.djl.Device;\n",
+    "import ai.djl.*;\n",
+    "import ai.djl.engine.Engine;\n",
+    "import ai.djl.metric.*;\n",
+    "import ai.djl.ndarray.*;\n",
+    "import ai.djl.ndarray.types.*;\n",
+    "import ai.djl.ndarray.index.*;\n",
+    "import ai.djl.nn.*;\n",
+    "import ai.djl.nn.core.*;\n",
+    "import ai.djl.training.*;\n",
+    "import ai.djl.training.initializer.*;\n",
+    "import ai.djl.training.loss.*;\n",
+    "import ai.djl.training.listener.*;\n",
+    "import ai.djl.training.evaluator.*;\n",
+    "import ai.djl.training.optimizer.*;\n",
+    "import ai.djl.training.optimizer.learningrate.*;\n",
+    "import ai.djl.training.dataset.*;\n",
+    "import ai.djl.util.*;\n",
+    "import java.util.Random;\n",
+    "import ai.djl.training.listener.TrainingListener;\n",
+    "import ai.djl.basicdataset.FashionMnist;\n",
+    "import ai.djl.training.dataset.Dataset;\n",
+    "import ai.djl.nn.norm.Dropout;\n",
+    "import tech.tablesaw.api.*;\n",
+    "import tech.tablesaw.plotly.api.*;\n",
+    "import tech.tablesaw.plotly.components.*;\n",
+    "import tech.tablesaw.plotly.Plot;\n",
+    "import tech.tablesaw.plotly.components.Figure;\n",
+    "import org.apache.commons.lang3.ArrayUtils;"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can test out the `dropoutLayer()` function on a few examples.\n",
+    "In the following lines of code, \n",
+    "we pass our input `X` through the dropout operation,\n",
+    "with probabilities 0, 0.5, and 1, respectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "NDManager manager = NDManager.newBaseManager();\n",
+    "\n",
+    "public NDArray dropoutLayer(NDArray X, float dropout) {\n",
+    "\n",
+    "    if (dropout == 1.0f) {\n",
+    "        return manager.zeros(X.getShape());\n",
+    "    }\n",
+    "\n",
+    "    if (dropout == 0f) {\n",
+    "        return X;\n",
+    "    }\n",
+    "\n",
+    "    NDArray mask = manager.randomUniform(0, 1, X.getShape());\n",
+    "    NDArray tempArray = mask.gt(dropout);\n",
+    "\n",
+    "    for (int i = 0; i < mask.size(); i++) {\n",
```

Comment:
What are you trying to achieve in here?

```
    mask = np.random.uniform(0, 1, X.shape) > dropout
```
Shouldn't be the tempArray as mask?

The boolean mask can multiply by X